### PR TITLE
feat: allow saving rmc/s runs

### LINKED ISSUE #93: add ability to save/load IP run for rmc/s/t
**Is your feature request related to a problem? Please describe.**
The plugin ends up in a bad state that prevents switching map.
The tmx api goes down.
The game crashes.
etc

**Describe the solution you'd like**
It would be nice if unexpected issues are happening, I could save the IP run and then come back to it later. This doesnt necessarily need to be a manual save either, it could just be an autosave when I finish a map of my current state. Then the UI could allow to load a previous run and continue it as if the problem never happened.

**Describe alternatives you've considered**
Manual vs Autosaving - what are the pros and cons of each? Id prefer a rolling backup that autosaved for unexpected issues to be the most reliable... but you could get by with a manual save in most scenarios.

Concerns around manipulation - sure, but i think ultimately video proof is required atm, so anybody faking times or medal counts would be verifiable. i think ultimately, any tracking of that data should ultimately be done server side for leaderboard automation.

**Additional context**
Just another alternative really when trackmania is being bad. I'd be willing to work on this, maybe later this week.

### PR DESCRIPTION
**Pull request by EdVraz**
resolves #93 

This seems to be working on my end but I'd appreciate further testing to ensure I didn't overlook anything.

Also, I would greatly appreciate it if we could get this in 2.3.5 already so that this does not hang unreleased for a month as, in my opinion, this is a very useful functionality.

### CODE REVIEW

File: src/Utils/MX/Methods/LoadRandomMap.as [hunk 1]
```diff
@@ -226,10 +226,28 @@ namespace MX
                 return;
             }
             RandomMapIsLoading = true;
-            if (preloadedMap is null && !isLoadingPreload) PreloadRandomMap();
-            while (isLoadingPreload) yield();
-            MX::MapInfo@ map = preloadedMap;
-            @preloadedMap = null;
+            MX::MapInfo@ map;
+            if (RMC::ContinueSavedRun && !RMC::IsInited) {
+                string url = "https://trackmania.exchange/api/maps/get_map_info/id/" + tostring(RMC::CurrentMapID);
```
Comment (GreepTheSheep) [inline src/Utils/MX/Methods/LoadRandomMap.as:231]: Please use `MX_URL` constant

File: src/Utils/RMC/RMC.as [hunk 2]
```diff
@@ -32,14 +32,26 @@ class RMC
...
                 RMC::StartTime = -1;
                 RMC::EndTime = -1;
                 @MX::preloadedMap = null;
+
+#if DEPENDENCY_CHAOSMODE
+                ChaosMode::SetRMCMode(false);
+#endif
+                int secondaryCount = RMC::selectedGameMode == RMC::GameMode::Challenge ? BelowMedalCount : RMC::Survival.Skips;
+                if (RMC::GoalMedalCount != 0 || secondaryCount != 0 || RMC::GotBelowMedalOnCurrentMap || RMC::GotGoalMedalOnCurrentMap) {
+                    Renderables::Add(SaveRunQuestionModalDialog());
```
Comment (GreepTheSheep) [inline src/Utils/RMC/RMC.as:48]: Can you add a Plugin Setting (in `src/Settings/RMCParameters.as`) to auto save run on exit and set it to true by default ? so we don't have to open this dialog window everytime we click on stop button

**Review by GreepTheSheep — CHANGES_REQUESTED** [review]

File: src/Utils/MX/Methods/LoadRandomMap.as (hunk 1 above)
Comment (EdVraz) [reply src/Utils/MX/Methods/LoadRandomMap.as:231]: Iirc this endpoint does not exist on https://map-monitor.xk.io/

Comment (GreepTheSheep) [reply src/Utils/MX/Methods/LoadRandomMap.as:231]: Then add a condition that uses `MX_URL` on MP4 and trackmania.exchange on TMNEXT

Comment (EdVraz) [reply src/Utils/MX/Methods/LoadRandomMap.as:231]: alright, also going to make an issue on the map-monitor repository for support of the endpoint

**Review by GreepTheSheep — APPROVED** [review]

**Pull request merged**